Modules/Scripted/Endoscopy/Endoscopy.py

```diff
@@ -707,28 +551,31 @@ class EndoscopyLogic:
 
     NODE_PATH_CAMERA_ORIENTATIONS_ATTRIBUTE_NAME = "Endoscopy.Path.CameraOrientations"
 
-    def __init__(self, inputCurve, cameraOrientations, dl=0.5):
-        self.cleanup()
-        self.cameraOrientations = cameraOrientations
+    def __init__(self, dl=0.5):
         self.dl = dl  # desired world space step size (in mm)
-        self.setControlPoints(inputCurve)
-
-    def cleanup(self):
-        # Whether we're about to construct or delete, free all resources and initialize class members to None.
-        self.dl = None
         self.inputCurve = None
-        self.resampledCurve = None
+
         self.planeNormal = None
-        self.cameraOrientations = None
+
+        self.resampledCurve = slicer.vtkMRMLMarkupsCurveNode()
         self.cameraOrientationResampledCurveIndices = None
 
-    def setControlPoints(self, inputCurve: slicer.vtkMRMLMarkupsCurveNode) -> bool:
+        self.updatingControlPoints = False
+
+    def getNumberOfControlPoints(self):
+        return self.resampledCurve.GetNumberOfControlPoints()
+
+    def setControlPoints(self, inputCurve: slicer.vtkMRMLMarkupsCurveNode) -> None:
+        if self.updatingControlPoints:
+            return
+
         expectedType = slicer.vtkMRMLMarkupsCurveNode
         if not isinstance(inputCurve, expectedType):
```

Comment:
If the user does "Close scene" while the Endoscopy module is still open, we get an immediate `TypeError` from here because the variable is now `None`.